pipelines/google/google_gemini.py

```diff
@@ -208,22 +239,17 @@ def get_google_models(self, force_refresh: bool = False) -> List[Dict[str, str]]
             self.log.debug("Using cached model list")
             return self._model_cache
             
-        self.validate_api_key()  # Ensure API key is validated before proceeding
         try:
-            # Ensure genai is configured before listing models
-            genai.configure(api_key=self.valves.GOOGLE_API_KEY.get_decrypted())
+            client = self._get_client()
             self.log.debug("Fetching models from Google API")
-
-            models = genai.list_models()
-            available_models = [
-                {
-                    "id": self.strip_prefix(model.name),
-                    "name": model.display_name,
-                }
-                for model in models
-                if "generateContent" in model.supported_generation_methods
-                and model.name.startswith("models/") # Ensure we only get standard models
-            ]
+            models = client.models.list()
+            available_models = []
+            for model in models:

```

Comment:
When using vertex, the models.list() returns model.name, but all other attributes are None

https://github.com/googleapis/python-genai/issues/679